### PR TITLE
Development

### PR DESCRIPTION
**Pull request by zmekonnen251**
### General requirements
- [x] There are [no linter errors](https://github.com/microverseinc/linters-config).
- [x] Used [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- [x] Professional documentation .
### Ruby requirements
- [x] Followed list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

### Project requirements

- [x] Created a new Rails app called 'hello-rails-react'.
- [x] Installed needed node dependencies as described in the previous lesson's tutorial.
- [x] Created a table for storing your messages and create 5 different greetings. 
- [x] Created an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
- [x] Created a static view that will be the root of your app.
- [x] Created the `App` component with react-router and render it in your static view.
- [x] Created the `Greeting` component that will display a greeting. Set it up as a route in your App component.
- [x] Created a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.

Thanks for reviewing my project.

### CODE REVIEW

**Review by GzimAsani — CHANGES_REQUESTED** [review]
Hi @zmekonnen251 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has linter errors, kindly fix that and ask for another code review

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  # Speed up commands on slow machines / big apps [https://github.com/rails/spring]
+  # gem "spring"
+end
+
+group :test do
+  # Use system testing [https://guides.rubyonrails.org/testing.html#system-testing]
+  gem 'capybara'
+  gem 'selenium-webdriver'
+  gem 'webdrivers'
+end
```
Comment (DeliceLydia) [inline Gemfile]: - [ ] For this project you were required to use `bundling-rails` gem

Comment (DeliceLydia) [reply Gemfile]: - [ ] Please add `babel-loader` in your `package.json` file

Comment (DeliceLydia) [reply Gemfile]: - [ ] I can see you configured `babel.config.js` please go ahead and work on `webpack.config.js`

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @zmekonnen251, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._
- [ ] To set up well your project refer to this [link](https://github.com/microverseinc/curriculum-rails/blob/main/connect-frontend-frameworks/lessons/rails_and_webpack.md)

### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: Gemfile (hunk 1 above)
Comment (zmekonnen251) [reply Gemfile]: Hello @DeliceLydia, The `jsbundling-rails` way is not working. I worked the other way of using webpacker. by using `gem webpacker` and `gem react-rails`.I have done it using this way by using webpack [this way](https://www.freecodecamp.org/news/how-to-create-a-rails-project-with-a-react-and-redux-front-end-8b01e17a1db/)

**Review by t-yanick — APPROVED** [review]
Hi @zmekonnen251, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Clean/Organized code

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**